addons/mrp/models/mrp_bom.py

```diff
@@ -245,14 +248,29 @@ def check_cycle(v, visited, recStack, graph):
             recStack[v] = False
             return False
 
+        def update_product_boms(product_boms: defaultdict, product_ids):
+            products = self.env['product.product'].browse(product_ids)
+            product_boms.update(self._get_product2bom(products, bom_type='phantom',
+                picking_type=picking_type or self.picking_type_id, company_id=self.company_id.id))
+            # Set missing keys to default value
```

Comment:
it is weird to use a defaultdict if it is to force the missing key for each product, maybe use a simple dict. Also if you want to keep your inner method, instantiate the product_boms before to avoid pass it in arguments (also for product_ids ???).
```python
product_boms = {}

def update_product_boms(product_ids):
    products = self.env['product.product'].browse(product_ids)
    product_boms.update(dict.fromkeys(products, False))
    product_boms.update(self._get_product2bom(products, bom_type='phantom', picking_type=picking_type or self.picking_type_id, company_id=self.company_id.id))
```

Reply:
Yeah sure, I also did it for product_ids so that `update_product_boms` does not take any argument anymore